### ISSUE TITLE
Log parameter names improvement for structured loggers

### ISSUE THREAD

**Issue by srollinet** [dotnet/orleans#5865]
A lot of log messages use the "old" String.Format syntax, with `{0}`  style parameters 

    Log.Info("Starting AsyncAgent {0} on managed thread {1}", Name, Thread.CurrentThread.ManagedThreadId)

When you use a structured logger, it results in 2 parameters named `0` and `1`, which is not a really useful information.

It can also cause indexation errors if you index the log messages. In my case I use Graylog to manage my logs and I see a lot of indexing errors because the same parameter with name `0` is not always of the same type. (int, string, ...)

This could be replaced by something like

    Log.Info("Starting AsyncAgent {AgentName} on managed thread {ThreadId}", Name, Thread.CurrentThread.ManagedThreadId)

The goal is maybe not to change it everywhere, but at least to improve the most common log messages in a first time and to improve the other logs over time.

**Comment by ReubenBond** [comment]
We've been using structured logging for new log messages but haven't gone back to convert most older messages yet. I was hoping for an analyzer to be released which would help to ensure that no errors were introduced during the conversion. Please feel free to PR individual fixes

**Comment by bddckr** [comment]
I've been looking for the same kind of analyzer. There's `Microsoft.Extensions.Logging.Analyzers` but it's [not shipping](https://github.com/aspnet/Extensions/blob/39d5064a04546c9acaa2871c4e07ece93d332dd7/src/Logging/Logging.Analyzers/src/Microsoft.Extensions.Logging.Analyzers.csproj#L11-L12) - https://github.com/aspnet/Extensions/issues/658.

**Comment by peter-perot** [comment]
Yeah, logging looks really weird:

![image](https://user-images.githubusercontent.com/15671569/68952130-9fd66f80-07bf-11ea-8fbc-0fba15b89c70.png)

Especially in those case where you use _log parameters_ (not `string.Format` or the `$""` syntax) to output bars:

https://github.com/dotnet/orleans/blob/c683fa8f2fddf310c2dfc3c9cf43db4fb80be1ad/src/Orleans.Core/Runtime/OutsideRuntimeClient.cs#L199

This results in some really weird output like this when using a structured logger like Serilog:

```text
 "----------" Started OutsideRuntimeClient with Global Client ID:
```

**Comment by MaximTkachenko** [comment]
Hi, is this issue still actual? If so I can fix it in particular project(s).

**Comment by ReubenBond** [comment]
Hi @MaximTkachenko, yes this issue is still relevant. There are many places where logging could be converted to the semantic style and contributions would be welcome. Any uses of those `.Error(...)`, `.Info(...)`, etc helper methods are candidates for changing to `.LogError(...)`, etc

**Comment by stylesm** [comment]
I'd like to pick this up if possible? May not get around to updating every single instance but will work through as many as I can.

**Comment by ReubenBond** [comment]
@stylesm Sure thing, and please feel free to work on it bit-by-bit (no need for one large PR)

**Comment by lowkay** [comment]
Has there been much movement on this?

We have transitioned from "normal" logging to structured logging in our codebase. There were so many log messages it was impractical to manually update them all. We opted to write a semantic logging analyzer with a best-effort auto-fix.
While I'm not sure we can make the analyzer public, I gave it a quick run over the Orleans master branch and found:
~ 300 log messages using numbered placeholders
~ 150 of those can be auto-fixed currently

There could be more than those 300, it was just a cursory look.

The rest need to be manually fixed, because the auto-fix can't come up with a good name (it does its best looking at variable names and property paths, but it bails if it encounters a complete expression or methods).

I'm happy to contribute the fixes, I need to find some time to do that though, if nobody else is actively working on it.
